### ISSUE TITLE
netdata-updater.sh script disables itself on kickstart-static64.sh installations

### ISSUE THREAD

**Issue by knatsakis** [netdata/netdata#7197]
##### Bug report summary
The netdata-updater.sh script disables automatic cron updates (disables itself), on kickstart-static64.sh installations

##### OS / Environment
kickstart-static64.sh installations.

##### Netdata version (ouput of `netdata -V`)
v1.18.1

##### Component Name
netdata-updater.sh

##### Steps To Reproduce
Install netdata using kickstart-static64.sh. Run the netdata-updater.sh script (/etc/cron.daily/netdata-updater). The netdata-updater script is removed from /etc/cron.daily

##### Expected behavior
The netdata-updater.sh script should remain under /etc/cron.daily. (It must use the REINSTALL_COMMAND variable from .environment)

**Comment by knatsakis** [comment]
This is closely related to #7039

**Comment by knatsakis** [comment]
Blocked by #7198

**Comment by knatsakis** [comment]
Consider reverting commit 6277b861d9948ec10ab40b09cd0c8e9e5201780b when this is fixed

**Comment by knatsakis** [comment]
@Ferroin @prologic we should not forget to revert 6277b861d9948ec10ab40b09cd0c8e9e5201780b when this is fixed.

**Comment by Ferroin** [comment]
Based on investigation, this is a result of the `netdata-updater.sh` script not supporting installs with the `kickstart-static64.sh` script.

I'm going to investigate getting this working correctly, but it will likely take a while.

**Comment by Kramerican** [comment]
I am now seeing this exact same behavior resurface. My Netdata installations auto-updated for a while, and now they are stuck on v1.19.0-460-gc92dd8ee

checking /etc/cron.daily reveals that the netdata updater job has been removed from there, once again ...

**Comment by ikarlo** [comment]
> I am now seeing this exact same behavior resurface. My Netdata installations auto-updated for a while, and now they are stuck on v1.19.0-460-gc92dd8ee
> 
> checking /etc/cron.daily reveals that the netdata updater job has been removed from there, once again ...

Also on v1.20 in my installations

**Comment by prologic** [comment]
This was fixed on Jan 10 2020 by @Ferroin  with #7704  -- Are you saying this didn't fix the problem? @Kramerican In your case I can understand that you likely have an old enough version that you would need to manually re-install to get the fix. In fact now that I think about it based on @ikarlo 's comment I _think_ this would affect every NetData install via `kickstart_static64.sh`; I will quickly verify this and whether this is still a problem...

**Comment by prologic** [comment]
I _can_ confirm this is still an issue:

```sh
prologic@Jamess-iMac
Thu Mar 26 11:50:32
~
 130
$ ssh 10.0.0.101
prologic@10.0.0.101's password:
Linux debian 4.19.0-6-amd64 #1 SMP Debian 4.19.67-2+deb10u2 (2019-11-11) x86_64

The programs included with the Debian GNU/Linux system are free software;
the exact distribution terms for each program are described in the
individual files in /usr/share/doc/*/copyright.

Debian GNU/Linux comes with ABSOLUTELY NO WARRANTY, to the extent
permitted by applicable law.
Last login: Thu Jan  2 07:38:32 2020 from 10.0.0.104
-bash: warning: setlocale: LC_ALL: cannot change locale (en_US.utf-8)
prologic@debian:~$
prologic@debian:~$ sudo su 0-
-bash: sudo: command not found
prologic@debian:~$ ^C
prologic@debian:~$
prologic@debian:~$ su -
Password:
root@debian:~#
root@debian:~# s /^C
root@debian:~#
root@debian:~# ls
root@debian:~#
root@debian:~# ls /home/prologic/
root@debian:~#
root@debian:~# bash <(curl -Ss https://my-netdata.io/kickstart-static64.sh)
-bash: curl: command not found
root@debian:~#
root@debian:~# apt-get install curl
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following additional packages will be installed:
  libcurl4
The following NEW packages will be installed:
  curl libcurl4
0 upgraded, 2 newly installed, 0 to remove and 0 not upgraded.
Need to get 596 kB of archives.
After this operation, 1,123 kB of additional disk space will be used.
Do you want to continue? [Y/n] y
Get:1 http://deb.debian.org/debian buster/main amd64 libcurl4 amd64 7.64.0-4 [332 kB]
Get:2 http://deb.debian.org/debian buster/main amd64 curl amd64 7.64.0-4 [264 kB]
Fetched 596 kB in 2s (281 kB/s)
Selecting previously unselected package libcurl4:amd64.
(Reading database ... 27392 files and directories currently installed.)
Preparing to unpack .../libcurl4_7.64.0-4_amd64.deb ...
Unpacking libcurl4:amd64 (7.64.0-4) ...
Selecting previously unselected package curl.
Preparing to unpack .../curl_7.64.0-4_amd64.deb ...
Unpacking curl (7.64.0-4) ...
Setting up libcurl4:amd64 (7.64.0-4) ...
Setting up curl (7.64.0-4) ...
Processing triggers for man-db (2.8.5-2) ...
Processing triggers for libc-bin (2.28-10) ...
root@debian:~#
root@debian:~# bash <(curl -Ss https://my-netdata.io/kickstart-static64.sh)
 --- Downloading static netdata binary: https://storage.googleapis.com/netdata-nightlies/netdata-latest.gz.run ---
[/tmp/netdata-kickstart-akz7aH]# curl -sSL --connect-timeout 10 --retry 3 https://storage.googleapis.com/netdata-nightlies/sha256sums.txt
 OK

[/tmp/netdata-kickstart-akz7aH]# curl -sSL --connect-timeout 10 --retry 3 https://storage.googleapis.com/netdata-nightlies/netdata-latest.gz.run
 OK

 --- Installing netdata ---
[/tmp/netdata-kickstart-akz7aH]# sh /tmp/netdata-kickstart-akz7aH/netdata-latest.gz.run -- --auto-update

  ^
  |.-.   .-.   .-.   .-.   .  netdata
  |   '-'   '-'   '-'   '-'   real-time performance monitoring, done right!
  +----+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+--->

  (C) Copyright 2017, Costa Tsaousis
  All rights reserved
  Released under GPL v3+

  You are about to install netdata to this system.
  netdata will be installed at:

                    /opt/netdata

  The following changes will be made to your system:

  # USERS / GROUPS
  User 'netdata' and group 'netdata' will be added, if not present.

  # LOGROTATE
  This file will be installed if logrotate is present.

   - /etc/logrotate.d/netdata

  # SYSTEM INIT
  This file will be installed if this system runs with systemd:

   - /lib/systemd/system/netdata.service

   or, for older Centos, Debian/Ubuntu or OpenRC Gentoo:

   - /etc/init.d/netdata         will be created


  This package can also update a netdata installation that has been
  created with another version of it.

  Your netdata configuration will be retained.
  After installation, netdata will be (re-)started.

  netdata re-distributes a lot of open source software components.
  Check its full license at:
  https://github.com/netdata/netdata/blob/master/LICENSE.md
Please type y to accept, n otherwise: y
Creating directory /opt/netdata
Verifying archive integrity...  100%   All good.
Uncompressing netdata, the real-time performance and health monitoring system  100%
 --- Deleting stock configuration files from user configuration directory ---
 --- Attempt to create user/group netdata/netadata ---
Adding netdata user group ...
[/opt/netdata]# groupadd -r netdata
 OK

Adding netdata user account with home /opt/netdata ...
[/opt/netdata]# useradd -r -g netdata -c netdata -s /usr/sbin/nologin --no-create-home -d /opt/netdata netdata
 OK

 --- Add user netdata to required user groups ---
Group 'docker' does not exist.
 FAILED  Failed to add netdata user to secondary groups

Group 'nginx' does not exist.
 FAILED  Failed to add netdata user to secondary groups

Group 'varnish' does not exist.
 FAILED  Failed to add netdata user to secondary groups

Group 'haproxy' does not exist.
 FAILED  Failed to add netdata user to secondary groups

Adding netdata user to the adm group ...
[/opt/netdata]# usermod -a -G adm netdata
 OK

Group 'nsd' does not exist.
 FAILED  Failed to add netdata user to secondary groups

Adding netdata user to the proxy group ...
[/opt/netdata]# usermod -a -G proxy netdata
 OK

Group 'squid' does not exist.
 FAILED  Failed to add netdata user to secondary groups

Group 'ceph' does not exist.
 FAILED  Failed to add netdata user to secondary groups

Group 'nobody' does not exist.
 FAILED  Failed to add netdata user to secondary groups

Group 'I2C' does not exist.
 FAILED  Failed to add netdata user to secondary groups

 --- Check SSL certificates paths ---
 --- Install logrotate configuration for netdata ---
[/opt/netdata]# cp system/netdata.logrotate /etc/logrotate.d/netdata
 OK

[/opt/netdata]# chmod 644 /etc/logrotate.d/netdata
 OK

 --- Telemetry configuration ---
You can opt out from anonymous statistics via the --disable-telemetry option, or by creating an empty file /opt/netdata/etc/netdata/.opt-out-from-anonymous-statistics

 --- Install netdata at system init ---
Installing systemd service...
[/opt/netdata]# cp system/netdata.service /lib/systemd/system/netdata.service
 OK

[/opt/netdata]# systemctl daemon-reload
 OK

[/opt/netdata]# systemctl enable netdata
Created symlink /etc/systemd/system/multi-user.target.wants/netdata.service -> /lib/systemd/system/netdata.service.
 OK

 --- Install (but not enable) netdata updater tool ---
Update script is located at /opt/netdata/usr/libexec/netdata/netdata-updater.sh

 --- Check if we must enable/disable the netdata updater tool ---
Adding to cron
Auto-updating has been enabled. Updater script linked to: /etc/cron.daily/netdata-updater

netdata-updater.sh works from cron. It will trigger an email from cron
only if it fails (it should not print anything when it can update netdata).

 --- creating quick links ---
[/opt/netdata]# ln -s bin sbin
 OK

[/opt/netdata/usr]# ln -s ../bin bin
 OK

[/opt/netdata/usr]# ln -s ../bin sbin
 OK

[/opt/netdata/usr]# ln -s . local
 OK

[/opt/netdata]# ln -s etc/netdata netdata-configs
 OK

[/opt/netdata]# ln -s usr/share/netdata/web netdata-web-files
 OK

[/opt/netdata]# ln -s usr/libexec/netdata netdata-plugins
 OK

[/opt/netdata]# ln -s var/lib/netdata netdata-dbs
 OK

[/opt/netdata]# ln -s var/cache/netdata netdata-metrics
 OK

[/opt/netdata]# ln -s var/log/netdata netdata-logs
 OK

[/opt/netdata/etc/netdata]# rm orig
 OK

[/opt/netdata/etc/netdata]# ln -s ../../usr/lib/netdata/conf.d orig
 OK

 --- fix permissions ---
[/opt/netdata]# chmod g+rx\,o+rx /opt
 OK

[/opt/netdata]# chown -R netdata:netdata /opt/netdata
 OK

 --- fix plugin permissions ---
[/opt/netdata]# chown root:netdata usr/libexec/netdata/plugins.d/apps.plugin
 OK

[/opt/netdata]# chmod 4750 usr/libexec/netdata/plugins.d/apps.plugin
 OK

[/opt/netdata]# chown root:netdata usr/libexec/netdata/plugins.d/ioping
 OK

[/opt/netdata]# chmod 4750 usr/libexec/netdata/plugins.d/ioping
 OK

[/opt/netdata]# chown root:netdata usr/libexec/netdata/plugins.d/cgroup-network
 OK

[/opt/netdata]# chmod 4750 usr/libexec/netdata/plugins.d/cgroup-network
 OK

[/opt/netdata]# chown root:netdata bin/fping
 OK

[/opt/netdata]# chmod 4750 bin/fping
 OK

Save install options
 --- starting netdata ---
 --- Restarting netdata instance ---

Stopping all netdata threads
[/opt/netdata]# stop_all_netdata
 OK

Starting netdata using command 'systemctl start netdata'
[/opt/netdata]# systemctl start netdata
 OK


  ^
  |.-.   .-.   .-.   .-.   .-.   .  netdata              .-.   .-.   .-.   .-
  |   '-'   '-'   '-'   '-'   '-'   is installed now!  -'   '-'   '-'   '-'
  +----+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+--->

[/opt/netdata]# chmod 0644 /opt/netdata/etc/netdata/netdata.conf
 OK

 OK

[/tmp/netdata-kickstart-akz7aH]# rm /tmp/netdata-kickstart-akz7aH/netdata-latest.gz.run
 OK

[/tmp/netdata-kickstart-akz7aH]# rm -rf /tmp/netdata-kickstart-akz7aH
 OK

root@debian:~#
root@debian:~# ls -lah /etc/cron
cron.d/       cron.daily/   cron.hourly/  cron.monthly/ crontab       cron.weekly/
root@debian:~# ls -lah /etc/cron
cron.d/       cron.daily/   cron.hourly/  cron.monthly/ crontab       cron.weekly/
root@debian:~# ls -lah /etc/cron
root@debian:~# ls -lah /etc/cron.daily/
total 36K
drwxr-xr-x  2 root root 4.0K Mar 26 11:56 .
drwxr-xr-x 69 root root 4.0K Mar 26 11:56 ..
-rwxr-xr-x  1 root root 1.5K May 29  2019 apt-compat
-rwxr-xr-x  1 root root  355 Dec 29  2017 bsdmainutils
-rwxr-xr-x  1 root root 1.2K Apr 19  2019 dpkg
-rwxr-xr-x  1 root root  377 Aug 29  2018 logrotate
-rwxr-xr-x  1 root root 1.1K Feb 10  2019 man-db
lrwxrwxrwx  1 root root   51 Mar 26 11:56 netdata-updater -> /opt/netdata/usr/libexec/netdata/netdata-updater.sh
-rwxr-xr-x  1 root root  249 Sep 28  2017 passwd
-rw-r--r--  1 root root  102 Oct 11 17:58 .placeholder
root@debian:~# /opt/netdata/usr/libexec/netdata/netdata-updater.sh
Entering /tmp/netdata-updater-QifQvN

  ^
  |.-.   .-.   .-.   .-.   .  netdata
  |   '-'   '-'   '-'   '-'   real-time performance monitoring, done right!
  +----+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+--->

  (C) Copyright 2017, Costa Tsaousis
  All rights reserved
  Released under GPL v3+

  You are about to install netdata to this system.
  netdata will be installed at:

                    /opt/netdata

  The following changes will be made to your system:

  # USERS / GROUPS
  User 'netdata' and group 'netdata' will be added, if not present.

  # LOGROTATE
  This file will be installed if logrotate is present.

   - /etc/logrotate.d/netdata

  # SYSTEM INIT
  This file will be installed if this system runs with systemd:

   - /lib/systemd/system/netdata.service

   or, for older Centos, Debian/Ubuntu or OpenRC Gentoo:

   - /etc/init.d/netdata         will be created


  This package can also update a netdata installation that has been
  created with another version of it.

  Your netdata configuration will be retained.
  After installation, netdata will be (re-)started.

  netdata re-distributes a lot of open source software components.
  Check its full license at:
  https://github.com/netdata/netdata/blob/master/LICENSE.md
Creating directory /opt/netdata
Verifying archive integrity...  100%   All good.
Uncompressing netdata, the real-time performance and health monitoring system  100%
Unknown option ' --auto-update'. Ignoring it.
 --- Attempt to create user/group netdata/netadata ---
Group 'netdata' already exists.
User 'netdata' already exists.
 --- Add user netdata to required user groups ---
Group 'docker' does not exist.
 FAILED  Failed to add netdata user to secondary groups

Group 'nginx' does not exist.
 FAILED  Failed to add netdata user to secondary groups

Group 'varnish' does not exist.
 FAILED  Failed to add netdata user to secondary groups

Group 'haproxy' does not exist.
 FAILED  Failed to add netdata user to secondary groups

User 'netdata' is already in group 'adm'.
Group 'nsd' does not exist.
 FAILED  Failed to add netdata user to secondary groups

User 'netdata' is already in group 'proxy'.
Group 'squid' does not exist.
 FAILED  Failed to add netdata user to secondary groups

Group 'ceph' does not exist.
 FAILED  Failed to add netdata user to secondary groups

Group 'nobody' does not exist.
 FAILED  Failed to add netdata user to secondary groups

Group 'I2C' does not exist.
 FAILED  Failed to add netdata user to secondary groups

 --- Check SSL certificates paths ---
 --- Install logrotate configuration for netdata ---
[/opt/netdata]# chmod 644 /etc/logrotate.d/netdata
 OK

 --- Telemetry configuration ---
You can opt out from anonymous statistics via the --disable-telemetry option, or by creating an empty file /opt/netdata/etc/netdata/.opt-out-from-anonymous-statistics

 --- Install netdata at system init ---
Installing systemd service...
[/opt/netdata]# cp system/netdata.service /lib/systemd/system/netdata.service
 OK

[/opt/netdata]# systemctl daemon-reload
 OK

[/opt/netdata]# systemctl enable netdata
 OK

 --- Install (but not enable) netdata updater tool ---
Update script is located at /opt/netdata/usr/libexec/netdata/netdata-updater.sh

 --- Check if we must enable/disable the netdata updater tool ---
You chose *NOT* to enable auto-update, removing any links to the updater from cron (it may have happened if you are reinstalling)

Removing cron reference: /etc/cron.daily/netdata-updater

 --- creating quick links ---
[/opt/netdata]# rm sbin
 OK

[/opt/netdata]# ln -s bin sbin
 OK

[/opt/netdata/usr]# rm bin
 OK

[/opt/netdata/usr]# ln -s ../bin bin
 OK

[/opt/netdata/usr]# rm sbin
 OK

[/opt/netdata/usr]# ln -s ../bin sbin
 OK

[/opt/netdata/usr]# rm local
 OK

[/opt/netdata/usr]# ln -s . local
 OK

[/opt/netdata]# rm netdata-configs
 OK

[/opt/netdata]# ln -s etc/netdata netdata-configs
 OK

[/opt/netdata]# rm netdata-web-files
 OK

[/opt/netdata]# ln -s usr/share/netdata/web netdata-web-files
 OK

[/opt/netdata]# rm netdata-plugins
 OK

[/opt/netdata]# ln -s usr/libexec/netdata netdata-plugins
 OK

[/opt/netdata]# rm netdata-dbs
 OK

[/opt/netdata]# ln -s var/lib/netdata netdata-dbs
 OK

[/opt/netdata]# rm netdata-metrics
 OK

[/opt/netdata]# ln -s var/cache/netdata netdata-metrics
 OK

[/opt/netdata]# rm netdata-logs
 OK

[/opt/netdata]# ln -s var/log/netdata netdata-logs
 OK

[/opt/netdata/etc/netdata]# rm orig
 OK

[/opt/netdata/etc/netdata]# ln -s ../../usr/lib/netdata/conf.d orig
 OK

 --- fix permissions ---
[/opt/netdata]# chmod g+rx\,o+rx /opt
 OK

[/opt/netdata]# chown -R netdata:netdata /opt/netdata
 OK

 --- fix plugin permissions ---
[/opt/netdata]# chown root:netdata usr/libexec/netdata/plugins.d/apps.plugin
 OK

[/opt/netdata]# chmod 4750 usr/libexec/netdata/plugins.d/apps.plugin
 OK

[/opt/netdata]# chown root:netdata usr/libexec/netdata/plugins.d/ioping
 OK

[/opt/netdata]# chmod 4750 usr/libexec/netdata/plugins.d/ioping
 OK

[/opt/netdata]# chown root:netdata usr/libexec/netdata/plugins.d/cgroup-network
 OK

[/opt/netdata]# chmod 4750 usr/libexec/netdata/plugins.d/cgroup-network
 OK

[/opt/netdata]# chown root:netdata bin/fping
 OK

[/opt/netdata]# chmod 4750 bin/fping
 OK

Save install options
 --- starting netdata ---
 --- Restarting netdata instance ---

Stopping all netdata threads
[/opt/netdata]# stop_all_netdata
 OK

Starting netdata using command 'systemctl start netdata'
[/opt/netdata]# systemctl start netdata
 OK


  ^
  |.-.   .-.   .-.   .-.   .-.   .  netdata              .-.   .-.   .-.   .-
  |   '-'   '-'   '-'   '-'   '-'   is installed now!  -'   '-'   '-'   '-'
  +----+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+--->

[/opt/netdata]# chmod 0644 /opt/netdata/etc/netdata/netdata.conf
 OK

Switching back to /root
root@debian:~# ls -lah /etc/cron.daily/
total 36K
drwxr-xr-x  2 root root 4.0K Mar 26 11:58 .
drwxr-xr-x 69 root root 4.0K Mar 26 11:56 ..
-rwxr-xr-x  1 root root 1.5K May 29  2019 apt-compat
-rwxr-xr-x  1 root root  355 Dec 29  2017 bsdmainutils
-rwxr-xr-x  1 root root 1.2K Apr 19  2019 dpkg
-rwxr-xr-x  1 root root  377 Aug 29  2018 logrotate
-rwxr-xr-x  1 root root 1.1K Feb 10  2019 man-db
-rwxr-xr-x  1 root root  249 Sep 28  2017 passwd
-rw-r--r--  1 root root  102 Oct 11 17:58 .placeholder
root@debian:~# pwd
/root
root@debian:~# cd /etc/cron.daily/
root@debian:/etc/cron.daily# ls -lah
total 36K
drwxr-xr-x  2 root root 4.0K Mar 26 11:58 .
drwxr-xr-x 69 root root 4.0K Mar 26 11:56 ..
-rwxr-xr-x  1 root root 1.5K May 29  2019 apt-compat
-rwxr-xr-x  1 root root  355 Dec 29  2017 bsdmainutils
-rwxr-xr-x  1 root root 1.2K Apr 19  2019 dpkg
-rwxr-xr-x  1 root root  377 Aug 29  2018 logrotate
-rwxr-xr-x  1 root root 1.1K Feb 10  2019 man-db
-rwxr-xr-x  1 root root  249 Sep 28  2017 passwd
-rw-r--r--  1 root root  102 Oct 11 17:58 .placeholder
root@debian:/etc/cron.daily# pwd
/etc/cron.daily
root@debian:/etc/cron.daily#
```

Reopening...

**Comment by prologic** [comment]
Oh I _know_ what's happening here! 🤦‍♂️

These are the important lines:

```sh
 --- Install (but not enable) netdata updater tool ---
Update script is located at /opt/netdata/usr/libexec/netdata/netdata-updater.sh

 --- Check if we must enable/disable the netdata updater tool ---
You chose *NOT* to enable auto-update, removing any links to the updater from cron (it may have happened if you are reinstalling)

Removing cron reference: /etc/cron.daily/netdata-updater
```

The updater and thus the `kickstart_static64.sh` which I _believe_ gets re-runs even on an update isn't smart enough to understand what "hey this is an update, leave existing things alone"; it just goes and clobbers over the top of a previous install that was already installed with "yes please enable auto updates".

Hmm 🤔

I _believe_ I can fix this...

**Comment by prologic** [comment]
This should now be fixed with #8507

**Comment by Kramerican** [comment]
Awesome - I'll manually update across the line and report back if I see this happening again

**Comment by prologic** [comment]
You'll have to wait for us to create and publish a new Makeself archive. I _think_ ;)

**Comment by Kramerican** [comment]
Alrighty - I'll give it a week or so and see if a new version is out by then :)

**Comment by prologic** [comment]
You should be able to update and get the fix in a day or so. Once #8507 is approved and merged a new nightly will get built with the fix :)

**Comment by prologic** [comment]
@Kramerican Were you able to successfully test this with the fixed merged and in nightlies from a few nights ago?

**Comment by Kramerican** [comment]
@prologic Hi sorry no I didn't have time to. I just now updated a couple of systems to `v1.20.0-291-g2bdd0109` which I will keep an eye on in the coming days and report back - However, I can already say now that after my manual update using `kickstart-static64.sh` I get the cron job set up OK in `/etc/cron.daily` - so things are looking promising :)

**Comment by prologic** [comment]
That's awesome news indeed! Another bug bite the dust :)

I will close this issue, please feel free to open a new issue if you discover new issues or something doesn't work as expected for you! Thanks for your help :)

**Comment by Kramerican** [comment]
Not so fast @prologic - a check today that there has been an auto-update to `v1.20.0-298-gd817d60b` - but when I check in /etc/cron.daily .... The Netdata cron job has been removed once again :(

**Comment by prologic** [comment]
> Not so fast @prologic - a check today that there has been an auto-update to `v1.20.0-298-gd817d60b` - but when I check in /etc/cron.daily .... The Netdata cron job has been removed once again :(

That's only possible if your `/etc/netdata/.environment` is somehow wrong :/

Can you paste its contents here?

Thanks! Sorry for the nuisance!

**Comment by Kramerican** [comment]
Sure - here is the content of that file:

```
# cat /opt/netdata/etc/netdata/.environment
# Created by installer
PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
CFLAGS="-static -O3"
LDFLAGS=""
NETDATA_PREFIX="/opt/netdata"
NETDATA_CONFIGURE_OPTIONS=""
NETDATA_ADDED_TO_GROUPS=" adm squid nobody"
INSTALL_UID="0"
NETDATA_GROUP="netdata"
REINSTALL_OPTIONS=""
RELEASE_CHANNEL="nightly"
IS_NETDATA_STATIC_BINARY="yes"
NETDATA_LIB_DIR="/opt/netdata/var/lib/netdata"
NETDATA_SELECTED_DASHBOARD="react"
```

**Comment by prologic** [comment]
As I suspected:

```
REINSTALL_OPTIONS=""
```

This will fix itself if you re-install; or modify it yourself and add `-u`. This should fix it for you :)

**Comment by Kramerican** [comment]
Hello

OK, doing once again `bash <(curl -Ss https://my-netdata.io/kickstart-static64.sh)` and now I have in my .environment file:

```
...
REINSTALL_OPTIONS=" --auto-update"
...
```

And I see the cronjob in `/etc/cron.daily` :O)

So now, in theory, everything should finally be OK?

**Comment by prologic** [comment]
Yes! Because the fix I put up correctly (_finaly_) passed the `REINSTALL_OPTIONS` options to the installer. The Makeself archive is both an installer and upgrader. So yes it should now finally work for you :) -- As well as my own systems where I installed the statically built NetData agent.

**Comment by Kramerican** [comment]
Awesome! Thank you for all your help with this :O)

**Comment by prologic** [comment]
You're welcome!

**Comment by Kramerican** [comment]
@prologic This keeps reverting for me - I discovered today that now all my installations are stuck at `v1.20.0-326-ge3b57cf0` and the cron job in `/etc/cron.daily` is gone yet again, and in my `/opt/netdata/etc/netdata/.environment` I have `REINSTALL_OPTIONS=""`

I'm about ready to pack it in with regards to Netdata auto updates :D

**Comment by ikarlo** [comment]
It would be very useful if there was an option to hide cron output while job running.  This also to avoid excessive e-mails.

**Comment by prologic** [comment]
> @prologic This keeps reverting for me - I discovered today that now all my installations are stuck at `v1.20.0-326-ge3b57cf0` and the cron job in `/etc/cron.daily` is gone yet again, and in my `/opt/netdata/etc/netdata/.environment` I have `REINSTALL_OPTIONS=""`
> 
> I'm about ready to pack it in with regards to Netdata auto updates :D

I can confirm this on my own static installed instances too :/ Something is a mis here; I'll reopen this issue.

**Comment by prologic** [comment]
> It would be very useful if there was an option to hide cron output while the job is running. This also to avoid excessive e-mails.
> 
> I am thinking of change Cron MAILTO variable to blank, for example.

Do you mind filing a separate issue and assigning @Ferroin and myself so we can track that as a separate feature? :) Thanks!

**Comment by prologic** [comment]
Tracking the final fix for this in #8782
